test/unit/dialects/mysql/query-generator.test.js

```diff
@@ -571,6 +571,15 @@ if (dialect === 'mysql') {
           },
           needsSequelize: true,
         },
+        {
+          arguments: ['myTable', { name: 'foo' }, undefined, { updateOnDuplicate: ['name'] }],
```

Comment:
Could you use a `literal()` as the value for `name` in this test?

Also, would you mind moving this test over to [this test file](https://github.com/sequelize/sequelize/blob/main/test/unit/query-interface/upsert.test.ts) ? It would be interesting to test how each dialect handles this (even if the solution for the other dialects is just to throw an error).

This way we can also remove the test duplication between mysql & mariadb

Reply:
Moved